src/store/plugins/fungibleTokens.js

```diff
@@ -220,10 +219,11 @@ export default (store) => {
     { immediate: true },
   );
 
-  store.watch(
-    () => aeNextAccountIdx.value,
+  // Fetch balances whenever new account is added
+  watch(
+    aeAccounts,
     async (val, oldVal) => {
-      if (val !== oldVal) {
+      if (val.length !== oldVal.length) {
```

Comment:
Please test this with a fresh new wallet and also when restoring a wallet. `oldval` could be undefined and accessing `.length` may lead to runtime errors`

Reply:
`aeAccunts` is based on regular `accounts` which always returns an array. If user didn't restored the accounts the array is empty.